### PR TITLE
feat: add compose workflow for local dev

### PR DESCRIPTION
**Pull request by cpendery**
## :memo: Description

Add a docker-compose workflow for local development and update the documentation to reflect that

Closes https://github.com/anchore/grype/issues/782

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -5,18 +5,27 @@ There are a few useful things to know before diving into the codebase. This proj
 ## Getting started
 
 After cloning do the following:
+
 1. run `make bootstrap` to download go mod dependencies, create the `/.tmp` dir, and download helper utilities.
 2. run `make` to run linting, tests, and other verifications to make certain everything is working alright.
 
 Checkout `make help` to see what other actions you can take.
 
+## Getting started [Docker]
```
Comment (wagoodman) [inline DEVELOPING.md:14]: nit: could we nest the `## Getting started` section?

```
## Getting started

### Native development
 (todays content)

### Docker development
(new content)
```

```suggestion
### Docker development
```

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      DOCKER_TLS_CERTDIR: /certs
+    networks:
+      - docker
+    volumes:
+      - docker-certs-ca:/certs
+      - docker-certs-client:/certs/client
+  grype:
+    build:
+      context: .
+      dockerfile: local_dev.Dockerfile
```
Comment (wagoodman) [inline docker-compose.yaml:16]: nit: could we rename this?
```suggestion
      dockerfile: dev.Dockerfile
```

**Comment by spiffcs** [comment]
I think this is a great addition for people who are developing on Windows and want a more common development environment.

I talked with @wagoodman about this PR and the biggest notes were that what we have in grype should also exist in syft so I'll go ahead and make that change

Also small comment on getting the `dev.Dockerfile` to build as my local.

`docker compose version === v2.6.0`

File: local_dev.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+FROM docker:latest
+
+ENV GO_VERSION=1.18.2
+ENV PATH=$PATH:/usr/local/go/bin:/usr/bin/env:$(go env GOPATH)/bin
```
[the remark this answers is not in the document]
Comment (cpendery) [reply local_dev.Dockerfile:4]: Looking into it now. I added that to support using Delve, and I must have forgotten to test the building

[the remark this answers is not in the document]
Comment (cpendery) [reply local_dev.Dockerfile]: Fixed the builds, thanks for catching that

**Review by spiffcs — APPROVED** [review]

**Pull request merged**